src/internal/components/cartesian-chart/responsive-text.tsx

```diff
@@ -0,0 +1,47 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { memo, useEffect, useRef } from 'react';
+
+interface ResponsiveTextProps {
+  x: number;
+  y: number;
+  className?: string;
+  children: string;
+  maxWidth: number;
+}
+
+export default memo(ResponsiveText);
+
+function ResponsiveText({ x, y, className, children, maxWidth }: ResponsiveTextProps) {
+  const textRef = useRef<SVGTextElement>(null);
+
+  // Determine the visible width of the text and if necessary truncate it until it fits.
+  useEffect(() => {
+    renderTextContent(textRef.current!, children, maxWidth);
+  }, [maxWidth, children]);
+
+  return (
+    <text ref={textRef} x={x} y={y} style={{ textAnchor: 'end' }} className={className}>
+      {children}
+    </text>
+  );
+}
+
+function renderTextContent(textNode: SVGTextElement, text: string, visibleWidth: number) {
+  let visibleLength = text.length;
```

Comment:
I find this name a bit confusing, would be more like `maxLength` or `maxVisibleLength`

Reply:
It is not max but current visible length. We start from the full length and make it smaller incrementally until finding the max fitting length. However, it becomes the max fitting length only at the last iteration.